ansible/vagrant/README.md

```diff
@@ -1,14 +1,52 @@
-## Vagrant deployer for Kubernetes Ansible
+## Welcome to the Vagrant Deployer for Kubernetes Ansible
 
-This deployer sets up a vagrant cluster and installs kubernetes with flannel on it.
+This deployer sets-up a Kubernetes cluster on Vagrant.
 
-## Before you start !
+## Before You Start
 
-You will need a functioning vagrant provider. Currently supported are openstack, libvirt, and virtualbox.
+Make sure you have [git installed](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git) and clone the contrib repo:
+```
+git clone https://github.com/kubernetes/contrib.git
+```
+
+[Install Vagrant](https://www.vagrantup.com/downloads.html) if it's not currently installed on your system.
+
+You will need a functioning [vagrant provider](https://www.vagrantup.com/docs/providers/). Currently supported providers are openstack, libvirt, and virtualbox. Vagrant comes with VirtualBox support by default. No matter what provider you choose, you need to install the OpenStack Vagrant plugin:
+
+```
+vagrant plugin install vagrant-openstack-provider --plugin-version ">= 0.6.1"
+```
+
+Vagrant uses Ansible to automate the Kubernetes deployment. Install Ansible (Mac OSX example):
+```
+sudo easy_install pip
+sudo pip install ansible
+```
+
+Reference [Ansible installation](http://docs.ansible.com/ansible/intro_installation.html) documentation for other platforms.
+
+The DNS kubernetes-addon requires python-netaddr. Install netaddr (Mac OSX example):
+
+```
+sudo pip install python-netaddr
+```
+
+Reference the [python-netaddr documentation](https://pythonhosted.org/netaddr/installation.html) for additional installation instructions.
+
+
+## Caveats
 
-## USAGE
+Vagrant (1.7.x) does not properly select a provider. You will need to manually specify the provider. Refer to the Provider Specific Information section for using the proper `vagrant up` command.
 
-In general all that should be needed it to run
+
+## Usage
+
+Due to [Issue 636](https://github.com/kubernetes/contrib/issues/636)], the Vagrant deployer requires the packageManager source_type. Update ~/contrib/ansible/group_vars/all.yml with the following:
```

Comment:
Agreed. I believe @rutsky and/or @stephenrlouie will be adding coreos support, so 636 will definitely need to get fixed.